### ISSUE TITLE
macOS 10.13.1 - Laptop: ALC295: no sound from recognised speakers (mic and headphones OK)

### ISSUE THREAD

**Issue by carlox97** [insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12#8]
Hi, I've patched AppleHDA.kext with with the "AppleHDA Patcher.app". I had kernel panic after installing patched AppleHDA.kext in S/L/E, even with "-f" bootflag. Now I use aDummyHDA.kext (installed correctly) + config.plist required edits + codec commander.  
Both Microphone and Headphone work very well, but no audio from speakers, even though they are recognised.

I tried lilu+AppleALC few days ago, same problem with layout 3, but it was fixed with layout 28 (even though the volume was lower than Windows). Overall it was okay until I had problems with lilu (kernel panics and broken battery status). 
Thanks for the help.

[My Laptop](https://support.hp.com/us-en/product/hp-pavilion-15-bc000-notebook-pc-series/10862166/model/12152413/manuals)

[Archive.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/1461927/Archive.zip)

**Comment by insanelydeepak** [comment]
Hi 
so you need support for ALC295 ? right

**Comment by carlox97** [comment]
Hi! Yes, I hope it's not the wrong section!

**Comment by insanelydeepak** [comment]
layout 13 , remove AppleALC and dummy or any audio kexts 


[AppleHDA-280.12-ALC295.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/1462365/AppleHDA-280.12-ALC295.zip)

**Comment by carlox97** [comment]
Removed aDummyHDA.kext (AppleALC + lilu were already removed...) & KextToPatch from Config.plist.

Installed correctly your AppleHDA in S/L/E with layout id 13 injected via config.plist but no audio devices at all.
If I use an HDAEnabler for layout 13, system crash on apple logo after verbose.

With mirone's HDA for ALC295 headphones and mic worked though, the problem is with the internal speakers...

**Comment by carlox97** [comment]
I'm quite new in hackintosh but I'm doing my researches to improve: I have EAPD 0x2 for Internal Speakers (node 0x14), and [This Guide](http://forum.osxlatitude.com/index.php?/topic/1946-complete-applehda-patching-guide/) mention:

_Note 5: EAPD existence
We have to look carefully at the output nodes like Speaker and Headphone, since in some codecs there is an External Amplifier(EAPD) to power up/down the Speaker to save power. We need to use EAPD command to wake up the node to get the sound otherwise we won't get sound even though its recognized._

**Comment by insanelydeepak** [comment]
so just install remove any patch kexts or aDummy kext or HDAEnabler , 

now just install kext i send and rebuild caches 

send you DSDT and Config.plist 

OSX version ??

**Comment by carlox97** [comment]
I've already done it... NO audio devices at all (with layout 13). I already removed what you mentioned (patch kexts, aDummy and HDAEnabler).

macOS 10.13.1...
DSDT and Config.plist sent in first post... Here are the files!
[Archive.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/1463706/Archive.zip)
Thanks.

**Comment by insanelydeepak** [comment]
use this one,
 report with trouble shoot files 

[SSDT-HDEF.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/1463749/SSDT-HDEF.zip)

**Comment by carlox97** [comment]
Now all devices recognised, but still no audio output from speakers (which works on Windows ofc).
Mic and Headphones are OK.

**Comment by insanelydeepak** [comment]
thats great 
only speaker is not working ? what about External Mic ?

**Comment by carlox97** [comment]
that's great? The problem is with speakers: headphones and internal mic works and worked with mirone's app too! It's the speakers that does not work, even if recognised in SysPrefs-Audio, no matter what.

I've just one device to test external mic and it can be broken... I'm okay with external mic not working, it's not a problem...

**Comment by insanelydeepak** [comment]
> The problem is with speakers: headphones and internal mic works and worked with mirone's app too! It's the speakers that does not work, even if recognised in SysPrefs-Audio, no matter what.

Hi im working from scratch so its a good news for me atleast , as far as your codec concen im not done yet 

it took much of time and patience to make perfect working codec 
so if you with me we can fix this too

**Comment by carlox97** [comment]
Yeah it's a good news for me too, but it was the same situation with mirone's app!
Ok I'm with you for sure! Thank you very much!

**Comment by insanelydeepak** [comment]
> but it was the same situation with mirone's app!
im fixing other one mess , just making my own :)

will try soon to fix it 

sens me
Screenshot of System Information/Hardware/Audio/Intel High Definition Audio

**Comment by carlox97** [comment]
Done!
[Audio.zip](https://github.com/insanelydeepak/Patched-AppleHDA-for-Mac-OS-Sierra-10.12/files/1463777/Audio.zip)

If you need codec dump or whatever else I can send that too! 
Thanks again.

**Comment by insanelydeepak** [comment]
are you sure your audio working?? 
it dont seems it working with these files

**Comment by carlox97** [comment]
Yeah It worked for sure from headphones and internal mic.
I fixed it by myself, my friend! I used mirone's AppleHDA.kext:

- In platforms.xml (decompressed) the nodes path for speakers were 23 (0x17)->3 (0x03), but from my codecdump my speakers have 20 (0x14)->2 (0x02) as nodes path.

- In PlugIns/AppleHDAHardwareConfigDriver.kext/info.plist I edited the verbs from "01771C20 01771D00 01771E17 01771F99" to "01471C20 01471D00 01471E17 01471F99". For some reason EAPD was present for node 0x14, even if node 0x14 was not present in verbs ("01470C02"), but I need it, so it's great!

Thank you very much regardless for the availability.

**Comment by insanelydeepak** [comment]
what do you mean ? you add EAPD and speakers works ??

**Comment by carlox97** [comment]
No. The AppleHDA generated by Mirone's tool used wrong verbs and nodes path for my speakers.
So I edited the nodes path to match my codec_dump (from 23->3 to 20->2) in platforms.xml.

Then I edited the verbs (PlugIns/AppleHDAHardwareConfigDriver.kext/info.plist) from 
`<01271c10 01271d00 01271ea6 01271f99 **01771c20 01771d00 01771e17 01771f99** 01971c30 01971d10 01971e81 01971f02 02171c40 02171d10 02171e21 02171f02 01770c02 **01470c02** 02170c02>`

to

`<01271c10 01271d00 01271ea6 01271f99 **01471c20 01471d00 01471e17 01471f99** 01971c30 01971d10 01971e81 01971f02 02171c40 02171d10 02171e21 02171f02 01770c02 **01470c02** 02170c02>`

If you look the mirone's verbs, you see that EAPD for node 0x14 is already present (01470c02), even though the node is not present in any previous verbs.

**Comment by insanelydeepak** [comment]
EAPD is only useful when audio not working after sleep , i will fix this later

**Comment by carlox97** [comment]
I'm not sure I understand what you mean, but I'll useEAPD; without codeccommander audio doesn't even work after wake... So I left that EAPD verbs (both for speakers and headphones, according to codecdump). You can close this issue if you feel! I thank you again for the help and for the AppleHDA patching guide, I wouldn't have been able to fix my speakers without it!

**Comment by insanelydeepak** [comment]
Hi
i was kinda busy  
your issue still exist ? we can fix it now ?

**Comment by anhbinhvodanh** [comment]
@insanelydeepak 
Can you help me with AppleHDA-276.26-ALC255_v2 on macOS 10.13
Thank you so much for your help.

**Comment by insanelydeepak** [comment]
Hi 
sure

**Comment by francklite** [comment]
> 
> 
> Yeah It worked for sure from headphones and internal mic.
> I fixed it by myself, my friend! Now also speakers work! I used mirone's AppleHDA.kext:
> 
>     * In Reasources/platforms.xml.zlib (decompressed to platforms.xml with "Zlib.app") the nodes path for speakers were 23 (0x17)->3 (0x03), but from my codecdump my speakers have 20 (0x14)->2 (0x02) as nodes path.
> 
>     * In PlugIns/AppleHDAHardwareConfigDriver.kext/info.plist I edited the verbs from "01771C20 01771D00 01771E17 01771F99" to "01471C20 01471D00 01471E17 01471F99". For some reason EAPD was already present for node 0x14 ("01470C02"), even if node 0x14 was not present itself in verbs, but I need it, so it's ok!
> 
> 
> Thank you very much regardless for the availability. Also, I thank you for the guide you published on OSXArena, since I followed that to understand better what was going on!

Hello there, I'm Francklite from Italy and I have similar problem on my Hackintosh (ASUS ROG Strix GL503VS: i7-7700HQ; HM175 Express Chipset; GTX 1070M; ALC295), initially everything was perfectly working even audio (Chipset, CPU, GPU NVIDIA, RAM etc), but after some time using Windows the speakers have no sound on both os windows and Mac (speakers harware tested by tester, working fine and fully recognized by both OS). I checked every single setting, update drivers in windows, audio seems to working perfectly, if I try to increase or decrease volume both os load the slider, but no audio from speakers as I tell u. Only using earphones I can get sound and Jack port is working properly (when plug off earphones OS load speakers as output). Microphone is working properly too in both OS.
So I found this post that problably gives me a real solution, but i don't have any copy of the original Audio codec dump, and I don't know how find the right verbs and nodes locations. Can u guys help me? Thanks in advance!

**Comment by carlox97** [comment]
> > Yeah It worked for sure from headphones and internal mic.
> > I fixed it by myself, my friend! Now also speakers work! I used mirone's AppleHDA.kext:
> > ```
> > * In Reasources/platforms.xml.zlib (decompressed to platforms.xml with "Zlib.app") the nodes path for speakers were 23 (0x17)->3 (0x03), but from my codecdump my speakers have 20 (0x14)->2 (0x02) as nodes path.
> > 
> > * In PlugIns/AppleHDAHardwareConfigDriver.kext/info.plist I edited the verbs from "01771C20 01771D00 01771E17 01771F99" to "01471C20 01471D00 01471E17 01471F99". For some reason EAPD was already present for node 0x14 ("01470C02"), even if node 0x14 was not present itself in verbs, but I need it, so it's ok!
> > ```
> > 
> > 
> > Thank you very much regardless for the availability. Also, I thank you for the guide you published on OSXArena, since I followed that to understand better what was going on!
> 
> Hello there, I'm Francklite from Italy and I have similar problem on my Hackintosh (ASUS ROG Strix GL503VS: i7-7700HQ; HM175 Express Chipset; GTX 1070M; ALC295), initially everything was perfectly working even audio (Chipset, CPU, GPU NVIDIA, RAM etc), but after some time using Windows the speakers have no sound on both os windows and Mac (speakers harware tested by tester, working fine and fully recognized by both OS). I checked every single setting, update drivers in windows, audio seems to working perfectly, if I try to increase or decrease volume both os load the slider, but no audio from speakers as I tell u. Only using earphones I can get sound and Jack port is working properly (when plug off earphones OS load speakers as output). Microphone is working properly too in both OS.
> So I found this post that problably gives me a real solution, but i don't have any copy of the original Audio codec dump, and I don't know how find the right verbs and nodes locations. Can u guys help me? Thanks in advance!

There are reports that the Win10 Realtek audio drivers prevent onboard audio from working properly and these problems may affect both macOS and Win10 (I could find you the reference, but it'd take some time, it was below a toleda's post on TonyMacx).
I would suggest you to boot into Win10, uninstall Realtek Audio Drivers, and [prevent them from being installed again using Hardware ID method](https://www.ghacks.net/2017/06/03/stop-windows-from-installing-drivers-for-specific-devices/) (Windows 10 Pro version needed). After doing this, try to perform an EC Reset and see. If you do this, you won't have full audio functionalities on Windows 10, and you might witness a loss in overall audio quality in Win10, since you will use the stock microsoft audio drivers. 

Otherwise consider doing a clean reinstall and then follow guides for configurations similar to yours.
https://www.tonymacx86.com/threads/asus-rog-strix-gl503vs-high-sierra.276851/

This guy has your same configuration, try to find informations about "(AppleALC + CodecCommander + JackFix)"  he mentions under Audio.

[ITA - comment] Guarda, fare un Hackintosh su un portatile è davvero dura, ti devi sporcare le mani, e lo devi fare davvero, anche se è decisamente più facile di tempo. Devi anche accettare che magari non tutto funzionerà al 100%, a meno che persone competenti non abbiano fatto DSDT & SSDTs Patches e kexts per il tuo computer specifico. Onestamente, effettuarlo su desktop sarebbe la scelta migliore, ma anche io ai tempi ho insistito molto, e alla fine ho ottenuto risultati soddisfacenti anche su laptop, e ad oggi perfetti su desktop. Buona fortuna! :)
